update-bot/wlmbots/lib/article_iterator.py

```diff
@@ -15,10 +15,8 @@ def __init__(self, callbacks, categories=None):
         self.articles_per_category_limit = 0
         self.log_every_n = 100
         self.callbacks = callbacks
-        if categories:
-            self.categories = categories
-        else:
-            self.categories = []
+        self.categories = [] if categories is None else categories
+        self._excluded_articles = {}
```

Comment:
Yes, I use dictionary internally for performance reasons. The good thing is that defining it as a property abstracts that implementation detail away.

It's probably overkill because I expect the list will not be longer than 100 articles max. But the lookup is done for every of the 9000+ articles so we might gain some milliseconds (and spare the tool labs server some CPU cycles) by doing it like this.